code/src/nuvla/connector/kubernetes.py

```diff
@@ -596,6 +614,385 @@ def reboot(self):
+
+        the_job_name = self.create_job_name("helm-ver-check")
+        helm_command = "'helm list -n default --no-headers'"
+        helm_version_result = self.run_helm_container(the_job_name, helm_command)
+
+        if helm_version_result.returncode == 0 and not helm_version_result.stderr:
+            helm_log_result = self.read_a_log_file(the_job_name)
+            current_version = json.loads(self.job.get('payload', '{}'))['current-version']
+            project_name = json.loads(self.job.get('payload', '{}'))['project-name']
+            result = self.check_target_release\
+                (helm_log_result, target_release, current_version) # no bother
+            if not self.check_project_name(helm_log_result, project_name):
+                return \
+                    "Project name does not match between helm on NulvaEdge and nuvla.io %s"\
+                        %(f'{project_name}'), 99
+            helm_update_job_name = self.create_job_name("helm-update")
+            helm_update_cmd = self.helm_gen_update_cmd_repo(target_release)
+            helm_update_result = self.run_helm_container(helm_update_job_name, helm_update_cmd)
+            log.info(f"Helm update result:\n {helm_update_result}")
+            result = result + "\n" + helm_update_result.stdout
+        else:
+            result = "The helm version command gave error \n %s"%(f'{helm_version_result.stderr}')
+
+        return result, helm_update_result.returncode
+
+    def helm_gen_update_cmd_deployment(self, target_release):
+        """
+        Generate the helm command that will run the update
+        target_release: the new chart version
+
+        This version generates a command based on the deployment repository
+        can be removed once the repository version is in production
+        """
+
+        install_params_from_payload = json.loads(self.job.get('payload', '{}'))
+        log.info(json.dumps(install_params_from_payload, indent=2))
+
+        modules = install_params_from_payload['config-files']
+        for module in modules:
+            log.debug("Found module: %s",module)
+
+        project_name = install_params_from_payload['project-name']
+        if "nuvlaedge-" in project_name:
+            log.debug(f"project name index : {len('nuvlaedge-')}")
+            project_uuid = project_name[len("nuvlaedge-"):]
+            log.debug(f"Found UUID : {project_uuid}")
+        peripherals = self.get_helm_peripherals(modules)
+        env_vars = self.get_env_vars_string(install_params_from_payload)
+        working_dir = self.get_working_dir(install_params_from_payload)
+        mandatory_args = \
+            " --set HOME=%s \
+            --set NUVLAEDGE_UUID=nuvlabox/%s \
+            --set kubernetesNode=$THE_HOST_NODE_NAME \
+            --set NUVLAEDGE_JOB_ENGINE_LITE_IMAGE=nuvladev/job-lite:issue-112-update-ne \
+            --set vpnClient=true"%(f'{working_dir}',f'{project_uuid}',)
```

Comment:
Why not using directly a f-string ?

Something like the following:
```python
mandatory_args = \
            f" --set HOME={working_dir} \
            --set NUVLAEDGE_UUID=nuvlabox/{project_uuid} \
            --set kubernetesNode=$THE_HOST_NODE_NAME \
            --set NUVLAEDGE_JOB_ENGINE_LITE_IMAGE=nuvladev/job-lite:issue-112-update-ne \
            --set vpnClient=true"
```
(to be tested)

Reply:
OK. Can give it a try with my local built version...